core/src/edu/cornell/gdiac/main/model/WindModel.java

```diff
@@ -257,23 +290,43 @@ protected void draw(GameCanvas canvas, Vector2 drawScale) {
       return;
     }
 
-    canvas.draw(
-        windPolyTexture,
-        windColor,
-        windSource.x * drawScale.x,
-        windSource.y * drawScale.y,
-        (windSide == WindSide.LEFT ? -1 : 1) * windLength * drawScale.x,
-        (windSide == WindSide.LEFT ? -1 : 1) * windBreadth * drawScale.y
-    );
-
-    // Draw particles
-    if (windParticles != null) {
-      for (WindParticleModel windParticle : windParticles) {
-        windParticle.draw(
-            canvas,
-            drawScale
-        );
-      }
+    // NOTE: Check FanModel:draw for more details
+    // NOTE: Can probably be factored out
+    v2Cache1.set((windSide == WindSide.LEFT ? -1 : 1) * 6, 6); // Apparently random offset is needed
+    v2Cache1.rotateRad(windRotation);
+
+    float offsetX = v2Cache1.x;
+    float offsetY = v2Cache1.y;
+
+    // Dimensions of wind assets according to design
+    float windWidth = 32;
+
+    for (int i = 1; i <= Math.floor(windLength * drawScale.x / 32); i++) {
+      float x = windSource.x * drawScale.x - offsetX;
+      float y = windSource.y * drawScale.y - offsetY;
+
+      // Middle left vector:
+      v2Cache1.set(-getWidth() * drawScale.x * 0.98f - (windSide == WindSide.LEFT ? -1 : 1) * i * windWidth * 0.98f, 0);
```

Comment:
Why are you multiplying by 0.98 here?

Reply:
Otherwise we get a small gap (1px) per animation "block"; probably should've put a comment next to it, too. The 0.98 shrinks the actual block length so that everything is more condensed and the gap is merged.